lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -298,4 +308,47 @@ public function getUpdateSchemaList(): array
 
         return $schemaTool->getUpdateSchemaSql($allMetadata, true);
     }
+
+    /** @return list<string> containing the found issues */
+    private function validatePropertiesTypes(ClassMetadataInfo $class): array
+    {
+        return array_values(
+            array_filter(
+                array_map(
+                    /** @param FieldMapping $fieldMapping */
+                    static function (array $fieldMapping) use ($class): string|null {
+                        $fieldName    = $fieldMapping['fieldName'];
+                        $propertyType = $class->reflFields[$fieldName]?->getType();
+
+                        // If the field type is not a built-in type, we cannot check it
+                        if (! Type::hasType($fieldMapping['type'])) {
+                            return null;
+                        }
+
+                        // If the property type is not a named type, we cannot check it
+                        if (! ($propertyType instanceof ReflectionNamedType)) {
+                            return null;
+                        }
+
+                        $metadataFieldType = (new DefaultTypedFieldMapper())->getBuiltInType(get_class(Type::getType($fieldMapping['type'])));
```

Comment:
Does the metadata not have a reference to the `Type` instance? Do we have to go through `Type::getType()`?

Reply:
Could the `DefaultTypedFieldMapper` be kept in memory as a `private` property? A bit wasteful to instantiate it all the time, but also no big deal.